### DISCUSSION TITLE
First contact with opengrok takes too long or ends up with "internal server error"

### DISCUSSION THREAD

**Discussion by ahmadbanijamali** [oracle/opengrok#4465] [General]
Hi, 
The opengrok is used for a list of >150 repos and the first contact with opengrok takes too much time to process or ends up with "internal server error". The next runs is quite fast and smooth. The issue happens when I do run the opengrok url for couple of days.

Is it a cache issue? tomcat bad configuration or conflict in tomcat versions?

**Comment by vladak** [comment]
anything interesting in Tomcat (I assume) logs ? Specifically the internal errors - they are usually caused by some sort of exception.

Also, what exactly do you mean by the 1st contact ? Is this after reindex or Tomcat startup ?

**Comment by ahmadbanijamali** [comment]
Nothing interesting. everything looks fine. As I mentioned, it works pretty fine in runs >1.

 The 1st contact means when I open the opengrok url after a while, I get that error. But when I refresh the url, the opengrok works perfectly fine.

**Comment by vladak** [comment]
Try to increase verbosity of Tomcat logging. It is strange that internal errors are not recorded anywhere.
